src/main/java/DanChat.java

```diff
@@ -0,0 +1,87 @@
+            String taskStatusIcon = task.getStatusIcon();
+            System.out.println("\t" + taskNumber + ". " + "[" + taskStatusIcon + "] " + taskName);
+            taskNumber++;
+        }
+    }
+    public static boolean isValidInteger(String str) {
+        try {
+            Integer.parseInt(str);
+            return true;
+        } catch (NumberFormatException e) {
+            return false;
+        }
+    }
+    public static void main(String[] args) {
+        String line = "=============================";
+        System.out.println(line);
+        System.out.println(" /\\_/\\  ");
+        System.out.println("( o.o ) " + " " + "Hello! I'm DanChat");
+        System.out.println(" > ^ <  " + " " + "What can I do for you?");
+        System.out.println(line);
+
+        String text;
+        Task[] tasks = new Task[100];
+        int taskCount = 0;
+
+        Scanner in = new Scanner(System.in);
+        text = in.nextLine();
+
+        while (true) {
+            String[] words = text.split(" ");
+            if (text.trim().equals("bye")) {
+                printMessage("Bye. Hope to see you again soon!");
+                break;
+            }
+            else if (text.trim().equals("list")) {
+                System.out.println(line);
+                printTaskList(tasks);
+                System.out.println(line);
+            } else if (words[0].equals("unmark") && words.length == 2 && isValidInteger(words[1])) {
+//                String[] words = text.split(" ");
+                int taskNumber = Integer.parseInt(words[1]);
+                if (taskNumber > taskCount) {
+                    printMessage("Sorry, there is no task with index " + taskNumber);
+                } else {
+                    Task changeTask = tasks[taskNumber - 1];
+                    changeTask.setDone(false);
+                    printMessage("Ok, I have marked this task as not done yet\n \t"
+                            + "[" + changeTask.getStatusIcon() + "] " + changeTask.getDescription());
+                }
+            } else if (words[0].equals("mark") && words.length == 2 && isValidInteger(words[1])) {
+//                String[] words = text.split(" ");
+                int taskNumber = Integer.parseInt(words[1]);
+                if (taskNumber > taskCount) {
+                    printMessage("Sorry, there is no task with index " + taskNumber);
+                } else {
+                    Task changeTask = tasks[taskNumber - 1];
+                    changeTask.setDone(true);
+                    printMessage("Nice! I have marked this task as done\n \t"
+                            + "[" + changeTask.getStatusIcon() + "] " + changeTask.getDescription());
+                }
+            }
+            else {
+                Task t = new Task(text);
+                tasks[taskCount] = t;
+                taskCount++;
+                printMessage("added: " + text);
+            }
+            Scanner nextIn = new Scanner(System.in);
```

Comment:
This line to create another Scanner object may not be necessary. In every iteration of your while loop, a new Scanner instance is created. This may result in unnecessary memory allocation.

Suggestion: Use the existing Scanner in declared at the beginning of main() instead: `text = in.nextLine();`
This way, System.in would be wrapped in only one Scanner instance for the entire program.